### ISSUE TITLE
Unrequested connections to Sentry service 

### ISSUE THREAD

**Issue by holtwick** [getsentry/sentry-cocoa#851]
_Platform:_
- [ ]  iOS
- [ ]  tvOS
- [x]  MacOS
- [ ]  watchOS

_Swift:_
- [x]  Yes, v5
- [ ]  No

_sentry-cocoa installed with:_
- [x] Swift Package Manager
- [ ] CocoaPods
- [ ] Carthage
- [ ] Manually

Version of sentry-cocoa:
**6.x.x**

---
I have the following issue:

Sentry always sends data to the service. I only want it to connect to Sentry when a crash is happening or any other event I issue. My apps are privacy concerned and this is a no go.

**Steps to reproduce:**

```swift
SentrySDK.start { options in
        options.dsn 
        options.debug = false // Enabled debug when first installing is always helpful
        options.logLevel = .error
        options.enableAutoSessionTracking = false

        options.beforeSend = { event in
            log.info("SentrySDK wants to send event \(event.message)")
            return event
        }

        options.beforeBreadcrumb = { breadcrumb in
            log.info("SentrySDK wants to send breadcrumb \(breadcrumb.category)")
            return breadcrumb
        }
}
```

**Actual result:**
```
◽️ 🔷 I|*    441.479 [SmartCompany] <SmartCompany.swift:28>   SentrySDK wants to send breadcrumb started 
2020-11-12 14:52:36.525694+0100 SmartApp[27678:814068] dnssd_clientstub ConnectToServer: connect() failed path:/var/run/mDNSResponder Socket:4 Err:-1 Errno:1 Operation not permitted
2020-11-12 14:52:36.526006+0100 SmartApp[27678:814068] [] nw_resolver_create_dns_service_locked [C1] DNSServiceCreateDelegateConnection failed: ServiceNotRunning(-65563)
2020-11-12 14:52:36.526520+0100 SmartApp[27678:814068] Connection 1: received failure notification
2020-11-12 14:52:36.526614+0100 SmartApp[27678:814068] Connection 1: failed to connect 10:-72000, reason -1
2020-11-12 14:52:36.526651+0100 SmartApp[27678:814068] Connection 1: encountered error(10:-72000)
2020-11-12 14:52:36.527742+0100 SmartApp[27678:814068] Task <3A64A680-A6BC-4BAF-A462-C40FD6DF9C71>.<1> HTTP load failed, 0/0 bytes (error code: -1003 [10:-72000])
2020-11-12 14:52:36.531245+0100 SmartApp[27678:814072] Task <3A64A680-A6BC-4BAF-A462-C40FD6DF9C71>.<1> finished with error [-1003] Error Domain=NSURLErrorDomain Code=-1003 "A server with the specified hostname could not be found." UserInfo={NSUnderlyingError=0x600000c74000 {Error Domain=kCFErrorDomainCFNetwork Code=-1003 "(null)" UserInfo={_kCFStreamErrorCodeKey=-72000, _kCFStreamErrorDomainKey=10}}, NSErrorFailingURLStringKey=https://o120938.ingest.sentry.io/api/5514224/envelope/, NSErrorFailingURLKey=https://o120938.ingest.sentry.io/api/5514224/envelope/, _kCFStreamErrorDomainKey=10, _kCFStreamErrorCodeKey=-72000, NSLocalizedDescription=A server with the specified hostname could not be found.}
2020-11-12 14:52:36.543205+0100 SmartApp[27678:814072] Sentry - Error:: Request failed: Error Domain=NSURLErrorDomain Code=-1003 "A server with the specified hostname could not be found." UserInfo={NSUnderlyingError=0x600000c74000 {Error Domain=kCFErrorDomainCFNetwork Code=-1003 "(null)" UserInfo={_kCFStreamErrorCodeKey=-72000, _kCFStreamErrorDomainKey=10}}, NSErrorFailingURLStringKey=https://o120938.ingest.sentry.io/api/5514224/envelope/, NSErrorFailingURLKey=https://o120938.ingest.sentry.io/api/5514224/envelope/, _kCFStreamErrorDomainKey=10, _kCFStreamErrorCodeKey=-72000, NSLocalizedDescription=A server with the specified hostname could not be found.}
```

**Expected result:**
No connection to Sentry or at least returning `nil` from `beforeBreadcrumb` or `befoeEvent` should stop any communication.

**Comment by holtwick** [comment]
This might have been a temporary issue. I cannot reproduce this anymore. `options.enableAutoSessionTracking = false` now works as expected.

**Comment by philipphofmann** [comment]
Thanks for the update @holtwick.

**Comment by holtwick** [comment]
@philipphofmann I figured out it only tries to send if a previous run did not set `options.enableAutoSessionTracking = false` and some calls seem to be in the queue. This is acceptable behavior, since I would keep it turned off anyways without explicit OK from the user. 

Thanks for the great framework and service BTW! I came from HockeyApp and am more than happy what Sentry offers to macOS and iOS devs.
